internal_packages/keybase/lib/keybase-user.cjsx

```diff
@@ -0,0 +1,237 @@
+        Associate Emails with Key
+      </span>
+      <ParticipantsTextField
+        field="to"
+        className="keybase-participant-field"
+        participants={ participants }
+        change={ @_onRecipientFieldChange } />
+      <button className="btn btn-toolbar" onClick={ @_onDone }>Done</button>
+    </div>
+
+  _onRecipientFieldChange: (contacts) =>
+    @setState(contacts)
+
+  _onDone: =>
+    @props.onPopoverDone _.pluck(@state.to, 'email')
+    Actions.closePopover()
+
+
+module.exports =
+class KeybaseUser extends React.Component
+  @displayName: 'KeybaseUserProfile'
+
+  @propTypes:
+    profile: React.PropTypes.object.isRequired
+
+  constructor: (props) ->
+    super(props)
+    # should the "new email" list component be an input, or a button?
+    @state = {inputEmail: false}
+
+  _matchKeys: (targetKey, keys) =>
+    # given a single key to match, and an array of keys to match from, returns
+    # a key from the array with the same fingerprint as the target key, or null
+    if not targetKey.key?
+      return null
+
+    key = _.find(keys, (key) =>
+      # not sure if the toString is necessary?
+      return key.key? and key.key.get_pgp_fingerprint().toString('hex') == targetKey.key.get_pgp_fingerprint().toString('hex')
+    )
+
+    if key == undefined
+      return null
+    else
+      return key
+
+  _importKey: =>
+    # opens a popover requesting user to enter 1+ emails to associate with a
+    # key - a button in the popover then calls _save to actually import the key
+    #popoverTarget = ReactDOM.findDOMNode(@refs.button).getBoundingClientRect()
+    popoverTarget = ReactDOM.findDOMNode(@refs.button).getBoundingClientRect()
+
+    Actions.openPopover(
+      <EmailPopover onPopoverDone={ @_popoverDone } />,
+      {originRect: popoverTarget, direction: 'left'}
+    )
+
+  _popoverDone: (addresses) =>
+    # closes the popover, saves a key if an email was entered
+    {profile} = @props
+    keybaseUsername = profile.keybase_user.components.username.val
+
+    if addresses.length < 1
+      # no email addresses added, nop
+      return
+    else
+      @_save(keybaseUsername, addresses[0])
+
+    if addresses.length > 1
+      # add any extra ddresses the user entered
+      _.each(addresses.slice(1), (address) =>
+        @_addEmail(address)
+      )
+
+  _save: (keybaseUsername, address) =>
+    # save/import a key from keybase
+    kb.getKey(keybaseUsername, (key) =>
+      if not key?
+        console.error "Unable to fetch key for #{keybaseUsername}"
+      else
+        PGPKeyStore.saveNewKey(address, key, true) # isPub = true
+    )
+    return
+
+  _delete: (event) =>
+    # delete a locally saved key
+    email = event.target.attributes['data-email'].value
+    keys = PGPKeyStore.pubKeys(email)
+    if keys? and keys.length == 1
+      key = keys[0]
+    else if keys.length > 1
+      # TODO key metadata objects need a unique ID
+      key = @_matchKeys(@props.profile.key, keys)
```

Comment:
Would it be safe to call `matchKeys` all the time, even if there's just one? Sort of silly, but I feel like having two code paths could mean we accidentally don't test `matchKeys` :-)